### PR TITLE
Add SetReplication job to replace Evict and Replicate

### PR DESCRIPTION
**Pull request by jja725**
### What changes are proposed in this pull request?

Add SetReplication job to replace Evict and Replicate. 

Merge evict and replicate since they are similar

### Please clarify why the changes are needed. For instance,
Make Evict and Replicate idempotent and use one unified API
### Does this PR introduce any user facing changes?

no

### CODE REVIEW

File: job/server/src/main/java/alluxio/master/job/plan/PlanTracker.java [hunk 1]
```diff
@@ -296,4 +299,21 @@ private synchronized boolean removeFinished() {
...
+      Set<Pair<String, Long>> activeJobs = mCoordinators.values().stream()
+          .filter(x -> x.getPlanInfo().getJobConfig() instanceof SetReplicaConfig)
+          .map(x -> ((SetReplicaConfig) x.getPlanInfo().getJobConfig()))
+          .map(x -> new Pair<>(x.getPath(), x.getBlockId())).collect(Collectors.toSet());
+      SetReplicaConfig config = (SetReplicaConfig) jobConfig;
+      String path = config.getPath();
+      long blockId = config.getBlockId();
+      Pair<String, Long> block = new Pair<>(path, blockId);
+      if (activeJobs.contains(block)) {
+        throw new JobDoesNotExistException(String.format(
```
Comment (jja725) [inline job/server/src/main/java/alluxio/master/job/plan/PlanTracker.java:314]: not sure we should create a new kind of exception or just use the existing one to keep it simple.

**Comment by yuzhu** [comment]
could you resolve the conflict and I will review again? thanks

**Comment by yuzhu** [comment]
Seems to have a build failure?

**Comment by jja725** [comment]
@jja725 add a new test failure to sheet, the other one seems known failure.

**Review by yuzhu — APPROVED** [review]

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Pull request merged**